app/controllers/application_controller.rb

```diff
@@ -42,4 +42,14 @@ def user_not_authorized(exception)
   def redirect_to_prior_flow(args = {})
     redirect_to stored_location_for(:user) || :root, args
   end
+
+  def store_full_location_for(resource_or_scope, location)
```

Comment:
I reckon something of this nature would be accepted upstream to devise. A thought for the future, perhaps :wink: